component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1248,8 +1248,34 @@ public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String
         HttpURLConnection httpConnection;
         boolean connection;
         String smsMessage = SMSOTPConstants.SMS_MESSAGE;
-        String encodedMobileNo = URLEncoder.encode(mobile, CHAR_SET_UTF_8);
-        smsUrl = smsUrl.replaceAll("\\$ctx.num", encodedMobileNo).replaceAll("\\$ctx.msg",
+        String[] headerArray;
+        String receivedMobileNumber = URLEncoder.encode(mobile, CHAR_SET_UTF_8);
+        HashMap<String, Object> headerElementProperties = new HashMap<>();
+        if (StringUtils.isNotEmpty(headerString)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Processing HTTP headers since header string is available");
+            }
+            headerArray = headerString.split(",");
+            for (String header : headerArray) {
+                String[] headerElements = header.split(":");
+                if (headerElements.length > 1) {
+                    headerElementProperties.put(headerElements[0], headerElements[1]);
+                } else {
+                    log.info("Either header name or value not found. Hence not adding header which contains " +
```

Comment:
Shall we make this a debug log?

Reply:
Fixed https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/101/commits/0e46118a6a81fff6522f71ba36ffa7689a64e26a